### ISSUE TITLE
Failure in `main` pipeline

### ISSUE THREAD

**Issue by timofurrer** [gitlabhq/terraform-provider-gitlab#1039]
We seem to have an issue in the acceptance tests running on `main` branch:

https://github.com/gitlabhq/terraform-provider-gitlab/runs/6124104979?check_suite_focus=true#step:7:217

```
=== RUN   TestAccGitlabProject_basic
    resource_gitlab_project_test.go:123: Step 1/19 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # gitlab_project.foo will be updated in-place
          ~ resource "gitlab_project" "foo" {
                id                                               = "73"
                name                                             = "foo-1835780171598199123"
                tags                                             = [
                    "bar",
                    "foo",
                ]
                # (64 unchanged attributes hidden)
        
              ~ container_expiration_policy {
                  ~ enabled     = false -> true
                    # (4 unchanged attributes hidden)
                }
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

It seems that the `container_expiration_policy` has a new value. Is that because we've enabled the container registry? Why didn't it fail earlier?

**Comment by timofurrer** [comment]
Also, it only seems to affect EE.

**Comment by timofurrer** [comment]
Okay, it seems to be a sporadic issue ... a retry passed ...

**Comment by PatrickRice-KSC** [comment]
I'll check this out, but I'd venture that since it seems to be saying there is a pending change on the container_expiration_policy and that it's sporatic, it's likely that the container expiration policy is a background apply similar to what we've seen on other resources and it needs a wait block to ensure it's consistent.

**Comment by timofurrer** [comment]
Yes, it's what I suspected, too.

It goes into the same direction of https://gitlab.com/gitlab-org/gitlab/-/issues/352299

**Comment by PatrickRice-KSC** [comment]
So I ran the same make script I wrote back when I was testing the previous fix to run the test until it failed, and over 1,000 tests completed successfully before my gitpod environment shut down 🙁 

That would seem to indicate if it _is_ a race condition, it's a _very_ rare one. I can still put in that fix, but I'm going to have a really hard time seeing if it actually fixes the issue. I'll give it another try to see if I can get it to repro.

**Comment by timofurrer** [comment]
I've seen it happening twice here on GitHub - and never locally. I'm removing the `p1` label and let's not rush this as it really seems to be rare ...

**Comment by PatrickRice-KSC** [comment]
> I've seen it happening twice here on GitHub - and never locally.

Yeah, something really weird going on here. I made it up over 5,000 invocations in GitPod with no failures...

**Comment by woz5999** [comment]
I think this is an actual bug related to the `container_expiration_policy`

Specifically, we were experimenting with enabling the policy and now cannot disable it via the terraform provider. It properly shows the diff (as above) but applying has no effect on the setting.
